### PR TITLE
Add rule "Age" to validation.

### PR DESCRIPTION
**Pull request by malukenho**
Add "Age" rule in favor of "MinimumAge".

Age rule works pretty much the same way “minimumAge”
worked but it optionally accepts a maximum age as second
argument.

See #103.

### Blockers for merge

- [ ] Add `age` rule on README.
- [ ] Add `age` rule on Validator docblocs.
- [ ] Remove `minimumAge` rule from Validator docblocs.
- [x] Modify README to reflect removal of `minimumAge` (point deprecation in favor of `age`).
- [x] Clean up unrelated commits and merges.
- [x] Make `age` compatible with `minimumAge`.

### CODE REVIEW

**Comment by henriquemoody** [comment]
I'm sorry by our late.

@malukenho, can you please rebase with our current master an update documentation (README.md and Validator docblocks)?

File: library/Rules/Age.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            if ($this->maxAge) {
+                $limit = new DateTime('now + '.$this->maxAge. ' year');
+                return $birthday > $input and $limit < $input;
+            }
+
+            return $birthday > $input;
+
+        } elseif (!is_string($input) || (is_null($this->format) && false === strtotime($input))) {
+            return false;
+        } else {
```
Comment (henriquemoody) [inline library/Rules/Age.php]: Avoid to use `else`.

File: library/Rules/Age.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            $age = ((date('Ymd') - date('Ymd', strtotime($input))) / 10000);
+
+            if ($this->maxAge) {
+                return $age >= $this->minAge and $age <= $this->maxAge;
+
+            }
+            return $age >= $this->minAge;
+        }
+}
```
Comment (henriquemoody) [inline library/Rules/Age.php]: Missin `    ` here.

File: library/Rules/MinimumAge.php [hunk 3]
```diff
@@ -1,36 +0,0 @@
-<?php
-namespace Respect\Validation\Rules;
-
-use DateTime;
-
-class MinimumAge extends AbstractRule
```
Comment (henriquemoody) [inline library/Rules/MinimumAge.php]: Do the same was done in [Vowels](https://github.com/Respect/Validation/blob/master/library/Rules/Vowels.php#L9-L10) rule, for example, them we can remove it in the next major release.

File: tests/library/Respect/Validation/Rules/Age.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Respect\Validation\Rules;
+
+use DateTime;
+
+class AgeTest extends \PHPUnit_Framework_TestCase
```
Comment (henriquemoody) [inline tests/library/Respect/Validation/Rules/Age.php]: Wrong file name, also wrong path, it must be `tests/Rules/AgeTest.php`

**Comment by henriquemoody** [comment]
@malukenho, I'm sorry to make you change it again, but we decided to keep BC between [minor versions](http://semver.org/), so just let `maximumAge()` as it is unless `age()` is totally compatible with `maximumAge()` rule, if that's the case you can just use inheritance on `MaximumAge`.

File: tests/library/Respect/Validation/Rules/Age.php (hunk 4 above)
Comment (malukenho) [inline tests/library/Respect/Validation/Rules/Age.php]: My fail, that need to be removed

**Comment by henriquemoody** [comment]
@malukenho, is there any progress on that issue?

**Comment by AndyWendt** [comment]
I could look into this if you like and get it fixed up.  Unless there is something else you would rather have me work on.

**Comment by henriquemoody** [comment]
If @malukenho wants to pass this to another one I'm ok with that, let's just wait for his answer.

**Comment by henriquemoody** [comment]
@AndyWendt, since @malukenho didn't answer our question and there are 21 days since the last interaction, I'd be glad if you finish it.

**Pull request closed without merging**